File: advanced/challenge-1/task1/server/models/bus.js
"use strict";
const { Model } = require("sequelize");
const bcrypt = require("bcryptjs");

module.exports = (sequelize, DataTypes) => {
  class Bus extends Model {
    static associate(models) {
      Bus.hasMany(models.Schedule, {
        foreignKey: "bus_id",
        as: "schedules",
      });
    }
  }
  Bus.init(
    {
      capacity: {
        type: DataTypes.INTEGER,
        allowNull: false,
      },

      model: {
        type: DataTypes.STRING,
        allowNull: false,
      },
      make: {
        type: DataTypes.STRING,
        allowNull: false,
      },

      associated_driver_id: {
        type: DataTypes.BIGINT,
        allowNull: false,
      },
    },
    {
      sequelize,
      freezeTableName: true,
      tableName: "buses",
      modelName: "Bus",
      paranoid: true,
    }
  );

  return Bus;
};

File: src/model/bus.py
import ormar

from src.model.metadata import metadata, database


class Bus(ormar.Model):
    class Meta:
        tablename = "bus"
        metadata = metadata
        database = database

    id: int = ormar.Integer(primary_key=True)
    capacity: int = ormar.Integer()
    model: str = ormar.String(max_length=255)
    make: str = ormar.String(max_length=255)
